src/tests/HDFFile.cxx

```diff
@@ -286,6 +286,91 @@ class T_CommandHandler : public testing::Test {
     H5Fclose(fid);
   }
 
+  static void write_attributes_at_top_level_of_the_file() {
+    using namespace FileWriter;
+    using std::array;
+    using std::vector;
+    using std::string;
+    MainOpt &main_opt = *g_main_opt.load();
+    {
+      rapidjson::Document cfg;
+      cfg.Parse(R""({})"");
+      main_opt.config_file = merge(cfg, main_opt.config_file);
+    }
+    rapidjson::Document json_command;
+    json_command.Parse(R""({
+      "cmd": "FileWriter_new",
+      "nexus_structure": {
+        "attributes": {
+          "some_top_level_int": 42,
+          "some_top_level_string": "Hello Attribute"
+        }
+      },
+      "file_attributes": {
+        "file_name": "tmp_write_top_level_attributes.h5"
+      },
+      "job_id": "832yhtwgfskdf"
+    })"");
+    auto cmd = json_to_string(json_command);
+    auto fname = get_string(&json_command, "file_attributes.file_name");
+    ASSERT_GT(fname.v.size(), 8);
+
+    FileWriter::CommandHandler ch(main_opt, nullptr);
+    Msg msg;
+    msg.data = (char *)cmd.data();
+    msg.size = cmd.size();
+    ch.handle(msg);
+    ASSERT_EQ(ch.file_writer_tasks.size(), (size_t)1);
+    {
+      string cmd(R""({
+        "recv_type": "FileWriter",
+        "cmd": "file_writer_tasks_clear_all",
+        "job_id": "832yhtwgfskdf"
+      })"");
+      ch.handle({(char *)cmd.data(), cmd.size()});
+    }
+
+    // Verification
+    auto fid = H5Fopen(string(fname).c_str(), H5F_ACC_RDONLY, H5P_DEFAULT);
+    ASSERT_GE(fid, 0);
+    {
```

Comment:
I would also suggest putting these in separate functions to improve readability, they could also be useful in other tests.